x/gamm/types/pool_asset.go

```diff
@@ -107,3 +107,12 @@ func PoolAssetsCoins(assets []PoolAsset) sdk.Coins {
 	}
 	return coins
 }
+
+func GetPoolAssetByDenom(assets []PoolAsset, denom string) (PoolAsset, bool) {
```

Comment:
Does it make sense to return a `PoolAsset` pointer to avoid making a copy and using extra stack space?